Assignment4/NoeGAnes-d21c003/Task07.py

```diff
@@ -0,0 +1,71 @@
+#Task 07: Querying RDF(s)
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+#TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL
+#RDFLib
+ns= Namespace("http://somewhere#")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s)
+  
+#SPARQL
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+from rdflib.plugins.sparql import prepareQuery
+q1 = prepareQuery('''
+  SELECT 
+    ?Subject
+  WHERE { 
+    ?Subject rdfs:subClassOf ns:Person. 
+  }
+  ''',
+  initNs = { "rdfs":RDFS , "ns":ns}
+)
+for r in g.query(q1):
+  print(r)
+
+
+#TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)
+#RDFLib
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
+  print(s)
```

Comment:
You are not listing instances from all subclasses of Person

Reply:
la verdad que no se muy bien cómo hacerlo, ¿hay que incluir que tienen que ser subclase de persona o cómo?

Reply:
Y muchas gracias por el feedback, me ayuda mucho

Reply:
Lo puedes hacer de dos maneras:
- La buena: usa una funcion recursiva que primero liste las subclases de una clase, liste los individuos de la misma, y se llame a si misma para cada subclase.
- La aceptable pero no tan buena: usa un bucle doble. El primero saca las subclases de person, el anidado saca las instancias de cada subclase.